### PR TITLE
[tune] Add requirements-dev.txt and update docs for contributing

### PR DESCRIPTION
**Pull request by timonbimon**
## What do these changes do?
Makes contributing to tune a little easier, by incorporating some of the hiccups I had when contributing for the first time.

- Updates the docs
- Adds a requirement-dev.txt with the needed packages to make the tests in tune pass (+ packages needed for autoformatting)

### CODE REVIEW

**Comment by timonbimon** [comment]
@richardliaw 

follow-up on the missing docs/requirements-dev.txt that I mentioned in my first PR :)

File: doc/source/tune-contrib.rst [hunk 1]
```diff
@@ -89,7 +94,7 @@ burden and speedup review process.
 Documentation should be documented in `Google style <https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html>`__ format.
 
 We also have tests for code formatting and linting that need to pass before merge.
-Install `yapf==0.23, flake8, flake8-quotes`. You can run the following locally:
+Install `yapf==0.23, flake8, flake8-quotes` (these are also in the requirements-dev.txt found in ``python/ray/tune``). You can run the following locally:
```
Comment (richardliaw) [inline doc/source/tune-contrib.rst]:
```suggestion
Install `yapf==0.23, flake8, flake8-quotes` (these are also in the `requirements-dev.txt` found in ``python/ray/tune``). You can run the following locally:
```

File: python/ray/tune/requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+flake8==3.7.7
+flake8-quotes==2.0.1
+gym==0.12.5
+opencv-python==4.1.0.25
+pandas==0.24.2
+requests==2.22.0
+tabulate==0.8.3
+tensorflow==1.13.1
+yapf==0.23.0
```
Comment (richardliaw) [inline python/ray/tune/requirements-dev.txt]: Can we only pin `yapf`? Any downsides otherwise?

**Review by richardliaw — COMMENTED** [review]
This is great! Just 2 questions.

File: python/ray/tune/requirements-dev.txt (hunk 2 above)
Comment (timonbimon) [reply python/ray/tune/requirements-dev.txt]: Yes, that makes sense! :)

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
Thanks for this!

**Pull request merged**